pkg/cmd/openshift-sdn/proxy.go

```diff
@@ -4,6 +4,7 @@ import (
 	"fmt"
 	"net"
 	"net/http"
+	"sync/atomic"
```

Comment:
The use of atomics rather than mutexes seems like premature optimization, particularly given that `OnServicesSynced` and `OnEndpointsSynced` are each called exactly once ever,. (And by using atomics rather than a mutex you need the trickiness with the extra `initialized` field, etc.)

Reply:
@danwinship there's enough locking going on in the hybrid proxy, iptables proxy, and userspace proxy that I don't want to add another level of it in this wrapper :(

Reply:
You don't need an atomic initialized field though. In the iptables proxier there's the complication of wanting to drop the mutex before calling `proxier.syncRunner.Run()`, but you don't have to worry about that here; all you want to do when initialization happens is `close(wph.waitChan)` and you can just do that from inside the mutex